src/lib.rs

```diff
@@ -147,9 +147,11 @@
 //! # }
 //! ```
 
-// required by quick_error or IronOxideErr
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::type_complexity)]
+// // required by quick_error or IronOxideErr
```

Comment:
side-note: I wonder if this is still required?
```suggestion
// required by quick_error or IronOxideErr
```

Reply:
I don't think they are necessary, which is why I was commenting them out and messed up the un-comment. But I didn't want to fully test it as part of this PR.